pkg/defaults/defaults.go

```diff
@@ -883,3 +889,17 @@ func getClusterPoolPullSpec(ctx context.Context, claim *api.ClusterClaim, hiveCl
 
 	return clusterImageSet.Spec.ReleaseImage, nil
 }
+
+func resolveCLIOverrideImage(architecture api.ReleaseArchitecture, version string) (*coreapi.ObjectReference, error) {
+	if architecture == "" || architecture == api.ReleaseArchitectureAMD64 {
+		return nil, nil
+	}
+	if version == "" {
+		return nil, errors.New("non-amd64 releases require a version to be configured")
+	}
+	// Should never happen™ but better than a NPD below when stripping everything after the Y stream
+	if len(version) < 3 {
```

Comment:
We probably want to re-use the major/minor parsing from `processVersionChannel()` in the official release code - IIRC we can't parse e.g. `4.3` as full SemVer so that's why the regex-based stuff exists. In any case, `[:3]` breaks on e.g. `4.10`